Src/FluentAssertions/Types/MethodInfoSelector.cs

```diff
@@ -27,8 +28,16 @@ public MethodInfoSelector(Type type)
         /// Initializes a new instance of the <see cref="MethodInfoSelector"/> class.
         /// </summary>
         /// <param name="types">The types from which to select methods.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="types"/> is <c>null</c>.</exception>
         public MethodInfoSelector(IEnumerable<Type> types)
         {
+            Guard.ThrowIfArgumentIsNull(types, nameof(types));
+
+            if (types.Any(t => t is null))
```

Comment:
🔧 Maybe add a dedicated `Guard` for that?

Reply:
Maybe do that in a separate PR, as it is also applicable for `PropertyInfoSelector`